### ISSUE TITLE
is chainer training speed inherently much slower than other framework such as caffe? 

### ISSUE THREAD

**Issue by sharpstill** [chainer/chainercv#373]
I recently use chainer to train faster R-CNN for object detection.
I modified chainercv source code because all my image are 512 x 512 , so my batch size is 10.
I am using one Tesla K40 GPU, but I found each second can only train 2 image.
I have 500 thousands of image, which means after train 20 epoch it will spend 1 mounth.
It is too slow for me. and my classmate told me he use caffe to train resnet or GoogLenet for classify,
the caffe training speed is 4 times quicker than chainer?
is my GPU to slow or chainer too slow or I must write some code to use cudnn? did cudnn automatically used?

**Comment by yuyu2172** [comment]
No, most of the libraries wrap cuDNN to execute heavy-load computation, and their speed should not differ by large margin.
K40 is much slower than some of the fastest GPUs, so I would recommend you to get better GPUs.
In my opinion, your dataset is too large for your computational environment.

On comparison with Caffe, I measured performance locally on my machine, and ChainerCV is slightly faster than Caffe implementation.
https://github.com/chainer/chainercv/tree/master/examples/faster_rcnn#speed


> did cudnn automatically used?
cuDNN is automatically used in Chainer if installed properly.


> the caffe training speed is 4 times quicker than chainer?
I used Chainer to train ResNet with ImageNet, and speed of training was pretty competitive.
We are planning to support a program to train classification networks soon with ChainerCV.
I suspect that your friend failed to tune some training parameters.
Execution time differs a lot based on the setting of MultiProcessIterator.
